spec/workflow/choice_rule_spec.rb

```diff
@@ -75,6 +75,15 @@
         end
       end
 
+      context "with a missing compare key" do
+        let(:payload) { {"Variable" => "$.foo", "Next" => "FirstMatchState"} }
+        let(:input) { {"foo" => "bar"} }
+
+        it "raises an exception" do
+          expect { subject }.to raise_exception(Floe::InvalidWorkflowError, "Data-test Expression Choice Rule must have a compare key")
+        end
+      end
```

Comment:
```ruby
      context "with an invalid compare key" do
        let(:payload) { {"Variable" => "$.foo", "InvalidCompare" => "$.bar", "Next" => "FirstMatchState"} }
        let(:input)   { {"foo" => 0, "bar" => 1} }

        it "fails" do
          expect { subject }.to raise_exception(Floe::InvalidWorkflowError)
        end
      end
```

Reply:
:+1: yeah I added two more tests one with a typo'd key which would have succeeded on the prior regex approach, and one testing if you have multiple compare keys